tools/lambda-promtail/lambda-promtail/s3.go

```diff
@@ -182,9 +203,19 @@ func parseS3Log(ctx context.Context, b *batch, labels map[string]string, obj io.
 				// NLB logs don't have .SSSSSSZ suffix. RFC3339 requires a TZ specifier, use UTC
 				match[1] += "Z"
 			}
-			timestamp, err = time.Parse(parser.timestampFormat, match[1])
-			if err != nil {
-				return err
+
+			if parser.timestampType == "string" {
+				timestamp, err = time.Parse(parser.timestampFormat, match[1])
+				if err != nil {
+					return err
+				}
+			} else if parser.timestampType == "unix" {
+				// convert to microseconds so that we only use one function
+				usec, err := toMicroseconds(match[1])
+				if err != nil {
+					return err
+				}
+				timestamp = time.UnixMicro(usec).UTC()
```

Comment:
we could simplify this to a switch statement, and include a default cause with some warning level logging